### ISSUE TITLE
SimdMaskArray::toInt overflows int return type for large N

### ISSUE THREAD

**Issue by olauq** [VcDevel/Vc#154]
Vc version / revision | Operating System | Compiler & Version | Compiler Flags | Assembler & Version | CPU
----------------------|------------------|--------------------|----------------|---------------------|----
                      |                  |                    |                |                     |

## Testcase
```cpp
```

**Comment by mattkretz** [comment]
Yeah, I guess. Another good reason not to exceed N = sizeof(int) * CHAR_BIT (i.e. 32 on x86). I won't change this for Vc 1.x anymore. But please don't use N > 32. I can't guarantee correctness or even useful performance for such values.

**Comment by olauq** [comment]
Odd, I provided more info when I submitted the issue, wonder where that went, anyway.

If > 32 is unsupported then maybe a (static) assert somewhere might be in order?

**Comment by mattkretz** [comment]
There was only the empty "more info template". Since it was empty I removed it.

Yes, maybe a `static_assert` is necessary in at least the `toInt` function.

**Comment by mattkretz** [comment]
The real solution seems to be that no function returning `int` should be part of `mask`. Instead it could return `std::bitset<N>`, which can scale to arbitrary number of bits. I'm currently implementing this for Vc 2.0
